backend/app/user_handler.go

```diff
@@ -283,59 +275,66 @@ func (h *Handler) RegisterHandler(c *gin.Context) {
 	if getErr != gorm.ErrRecordNotFound {
 		if !user.Verified {
 			user.ID = existingUser.ID
-			if err := h.db.UpdateUserByID(&user); err != nil {
+			err = h.db.UpdateUserByID(&user)
+			if err != nil {
 				log.Error().Err(err).Send()
 				InternalServerError(c)
 				return
 			}
 		}
-		Success(c, http.StatusCreated, "Verification code sent successfully", RegisterResponse{
-			Email:   request.Email,
-			Timeout: fmt.Sprintf("%d seconds", h.config.MailSender.Timeout),
-		})
-		return
 	}
 
-	mnemonic, _, err := h.setupTFChainUser()
-	if err != nil {
-		log.Error().Err(err).Msg("failed to setup user on TFChain")
-		InternalServerError(c)
-		return
-	}
-	stripeID, err := createStripeCustomer(request.Name, request.Email)
-	if err != nil {
-		log.Error().Err(err).Send()
-		InternalServerError(c)
-		return
-	}
-	user.Mnemonic = mnemonic
-	user.StripeCustomerID = stripeID
-	sponseeKeyPair, err := sr25519.Scheme{}.FromPhrase(mnemonic, "")
-	if err != nil {
-		Error(c, http.StatusInternalServerError, "failed to create keypair for SS58 address", err.Error())
-		return
-	}
-	sponseeAddress, err := sponseeKeyPair.SS58Address(42)
-	if err != nil {
-		Error(c, http.StatusInternalServerError, "failed to get SS58 address", err.Error())
-		return
-	}
-	user.Address = sponseeAddress
-	if err := h.db.RegisterUser(&user); err != nil {
-		log.Error().Err(err).Send()
-		InternalServerError(c)
-		return
-	}
-	if err := h.createKYCSponsorship(mnemonic); err != nil {
-		log.Error().Err(err).Msg("failed to create KYC sponsorship")
-		InternalServerError(c)
-		return
-	}
-	// Refresh and cache verification status
-	if err := h.refreshUserVerification(&user); err != nil {
-		log.Error().Err(err).Msg("failed to refresh user verification after registration")
-		// Not fatal for registration, just log
+	// create user model in db
+	if getErr != nil {
+		mnemonic, _, err := internal.SetupUserOnTFChain(h.substrateClient, h.config)
+		if err != nil {
+			log.Error().Err(err).Msg("failed to setup user on TFChain")
+			InternalServerError(c)
+			return
+		}
+		customer, err := internal.CreateStripeCustomer(request.Name, request.Email)
+		if err != nil {
+			log.Error().Err(err).Send()
+			InternalServerError(c)
+			return
+		}
+
+		user.Mnemonic = mnemonic
+		user.StripeCustomerID = customer.ID
+		// Set user.Address from mnemonic
+		sponseeKeyPair, err := sr25519.Scheme{}.FromPhrase(mnemonic, "")
+		if err != nil {
+			log.Error().Err(err).Msg("failed to create keypair for SS58 address")
+			InternalServerError(c)
+			return
+		}
+		sponseeAddress, err := sponseeKeyPair.SS58Address(42)
+		if err != nil {
+			log.Error().Err(err).Msg("failed to get SS58 address")
+			InternalServerError(c)
+			return
+		}
+		user.Address = sponseeAddress
+
+		err = h.db.RegisterUser(&user)
+		if err != nil {
+			log.Error().Err(err).Send()
+			InternalServerError(c)
+			return
+		}
+
+		if err := h.createKYCSponsorship(mnemonic); err != nil {
```

Comment:
I think this should be done before user registration

Reply:
I suggest this to be part of the workflow after we return the user id and the workflow id we have the KYC sponsorship as part of the flow